src/lib/Builder.hs

```diff
@@ -135,6 +135,7 @@ emitAtomToName x = emit (Atom x)
 class (EnvReader m, MonadFail1 m) => HoistingTopBuilder m where
   emitHoistedEnv :: (SinkableE e, SubstE Name e, HoistableE e)
                  => Abs TopEnvFrag e n -> m n (Maybe (e n))
+  willItHoist :: HoistableE e => e n -> m n Bool
```

Comment:
`canHoistToTop`? It's much more precise, since `willItHoist` can always return `True`: `e n` always hoists to `n` :)